tests/integration/sync/test_sync_command.py

```diff
@@ -0,0 +1,398 @@
+import os
+
+import logging
+import click
+import json
+# import requests
+import shutil
+import time
+# import pytest
+from pathlib import Path
+from unittest import skipIf
+
+import boto3
+from botocore.exceptions import ClientError
+import docker
+from botocore.config import Config
+from parameterized import parameterized
+
+from samcli.lib.bootstrap.bootstrap import SAM_CLI_STACK_NAME
+from samcli.lib.bootstrap.companion_stack.data_types import CompanionStack
+from samcli.lib.config.samconfig import DEFAULT_CONFIG_FILE_NAME
+from samcli.lib.utils.resources import (
+    AWS_APIGATEWAY_RESTAPI,
+    # AWS_APIGATEWAY_V2_API,
+    AWS_LAMBDA_FUNCTION,
+    AWS_STEPFUNCTIONS_STATEMACHINE,
+)
+from tests.integration.buildcmd.build_integ_base import BuildIntegBase
+from tests.integration.sync.sync_integ_base import SyncIntegBase
+from tests.integration.package.package_integ_base import PackageIntegBase
+
+from tests.testing_utils import RUNNING_ON_CI, RUNNING_TEST_FOR_MASTER_ON_CI, RUN_BY_CANARY
+from tests.testing_utils import run_command_with_input
+
+# Deploy tests require credentials and CI/CD will only add credentials to the env if the PR is from the same repo.
+# This is to restrict package tests to run outside of CI/CD, when the branch is not master or tests are not run by Canary
+SKIP_SYNC_TESTS = RUNNING_ON_CI and RUNNING_TEST_FOR_MASTER_ON_CI and not RUN_BY_CANARY
+CFN_SLEEP = 3
+API_SLEEP = 5
+SFN_SLEEP = 10
```

Comment:
nit: Can we retry 10 times with a period of 1 second instead of waiting for 10 seconds all the time? 
We can then set longer durations like 20 seconds in case of slow API while having shorter testing time on average.

Reply:
Good point! Added a logic for this